sql/core/src/test/scala/org/apache/spark/sql/UDFSuite.scala

```diff
@@ -93,9 +96,22 @@ class UDFSuite extends QueryTest with SharedSQLContext {
     assert(sql("SELECT strLenScala('test')").head().getInt(0) === 4)
   }
 
-  test("ZeroArgument UDF") {
-    spark.udf.register("random0", () => { Math.random()})
```

Comment:
Even our test case does not follow our assumption. We do not expect users to define non-deterministic UDF before this PR.